editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-top-level-elements.ts

```diff
@@ -0,0 +1,55 @@
+import { useContextSelector } from 'use-context-selector'
+import { TopLevelElement } from '../../../core/shared/element-template'
+import { isParseSuccess, isTextFile } from '../../../core/shared/project-file-types'
+import { getContentsTreeFileFromString, ProjectContentTreeRoot } from '../../assets'
+import {
+  DerivedState,
+  EditorState,
+  getFileForName,
+  getOpenUIJSFileKey,
+  TransientFileState,
+} from '../../editor/store/editor-state'
+import { useEditorState } from '../../editor/store/store-hook'
+import { UtopiaProjectContext } from './ui-jsx-canvas-contexts'
+
+export function getTopLevelElementsFromEditorState(
+  filePath: string,
+  editor: EditorState,
+  derived: DerivedState,
+): TopLevelElement[] {
+  return getTopLevelElements(
+    filePath,
```

Comment:
Shouldn't this only rely on `getOpenUIJSXFileKey` and not `filePath` at all if it's for the canvas?

Reply:
Ah no, I see, this could be doing a lookup for the elements used by other elements.